djconnectwise/sync.py

```diff
@@ -2827,15 +2837,15 @@ class AgreementSynchronizer(Synchronizer):
     }
 
     def _assign_field_data(self, instance, json_data):
-        instance.id = json_data['id']
-        instance.name = json_data['name']
-        instance.agreement_type = json_data['type']['name']
+        instance.id = json_data.get('id')
+        instance.name = json_data.get('name')
+        instance.agreement_type = json_data.get('type').get('name')
```

Comment:
Same as above about calling `.get()` on output of `.get()`. Brings us back to the same problem.